cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/upscale/AzureUpscaleService.java

```diff
@@ -79,7 +78,7 @@
             List<CloudResource> templateResources = azureCloudResourceService.getDeploymentCloudResources(templateDeployment);
             List<CloudResource> newInstances =
                     azureCloudResourceService.getInstanceCloudResources(stackName, templateResources, scaledGroups, resourceGroupName);
-            List<CloudResource> osDiskResources = azureCloudResourceService.getAttachedOsDiskResources(ac, newInstances, resourceGroupName);
+            List<CloudResource> osDiskResources = azureCloudResourceService.getAttachedOsDiskResources(newInstances, resourceGroupName, client);
```

Comment:
The functionality of getting the instances and then the attached os disk resources is present at several places. It might make sense to extract it to a common method at a separate class.

Reply:
I see only 2 occurrences of a 2 line common fragment, I would suggest refactoring it once we see more later.